dipy/core/gradients.py

```diff
@@ -244,3 +244,55 @@ def gradient_table(bvals, bvecs=None, big_delta=None, small_delta=None,
                                            small_delta=small_delta,
                                            b0_threshold=b0_threshold,
                                            atol=atol)
+
+
+def reorient_bvecs(gtab, rots):
+    """
+    Reorient the directions in a GradientTable according
+
+    When correcting for motion, rotation of the diffusion-weighted volumes
+    might cause systematic bias in rotationally invariant measures, such as FA
+    and MD, and also cause characteristic biases in tractography, unless the
+    gradient directions are appropriatel reoriented to compensate for this
+    effect [Leemans2009]_.
+
+    Parameters
+    ----------
+    gtab : a GradientTable class instance.
+        The nominal gradient table with which the data were acquired.
+    rots : (n, 3) array.
```

Comment:
My understanding is that rotation angles are not well defined, different rotation angles can be used to represent the same transformation and different conventions exist for different applications. For example Euler angles vs Tait-Bryan angles.

Is there a way to simplify this, for example would it be easier to use an affine matrix or a "transformation object" to help standardize this?